docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,120 @@
 .. Copyright (c) 2020 The DAML Authors. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-Application architecture guide
-#########################################
+Application architecture
+########################
 
-This document is a guide to building applications that interact with a DA ledger deployment (the 'ledger'). It:
+This section describes our recommended design of a full-stack DAML application.
 
-- describes the characteristics of the ledger API, how this affects the way an application is built (the 'application architecture'), and why it is important to understand this when building applications
-- describes the resources in the SDK to help with this task
-- gives some guidelines to help you build correct, performant, and maintainable applications using all of the supported languages
+.. image:: ./recommended_architecture.svg
 
-Categories of application
-*************************
+The presented architecture is by no means your only option, the DAML SDK provides support libraries
+if you choose to develop your application in :ref:`Java <java-bindings>` and :ref:`Scala
+<scala-bindings>`, or if you wish to develop directly against a :ref:`gRPC <grpc>` endpoint instead
+of the JSON REST API.
 
-Applications that interact with the ledger normally fall into four categories:
+To get you started quickly with the recommended application architecture, clone the
+``create-daml-app`` application template:
 
-.. list-table:: Categories of application
-   :header-rows: 1
+.. code-block:: bash
 
-   * - Category
-     - Receives transactions?
-     - Sends commands?
-     - Example
-   * - Source
-     - No
-     - Yes
-     - An injector that reads new contracts from a file and injects them into the system.
-   * - Sink
-     - Yes
-     - No
-     - A reader that pipes data from the ledger into an SQL database.
-   * - Automation
-     - Yes
-     - Yes, responding to transactions
-     - Automatic trade registration.
-   * - Interactive
-     - Yes (and displays to user)
-     - Yes, based on user input
-     - DA’s :doc:`Navigator </tools/navigator/index>`, which lets you see and interact with the ledger
+  git clone https://github.com/digital-asset/create-daml-app
 
-Additionally, applications can be written in two different styles:
+``create-daml-app`` is a small, but fully functional demo application implementing the recommended
+architecture, including
 
--  Event-driven - applications base their actions on individual ledger events only.
--  State-driven - applications base their actions on some model of all contracts active on the ledger.
+- DAML React libraries
+- quick iteration against a sandbox backend
+- deployment in the cloud or on `DABL <https://projectdabl.com>`_
+- authentication
```

Comment:
```suggestion
- DAML React libraries
- quick iteration against against the :ref:`DAML Ledger Sandbox<sandbox-manual>`
- authentication
- deployment in the cloud or on `DABL <https://projectdabl.com>`_
```
How is "deployment in the cloud" covered by the template? Do we provide scripts for automatically deploying to DABL and another cloud provider?

Reply:
it contains a docker file. we want to make the creation of the docker image a bit easier as a next step. also the part about deployment on DABL has yet to come.

Reply:
Maybe then we should write something like this:

"DAR file that can be easily deployed to DABL as well as a Docker image that can be deployed to a cloud environment of your choice"

Reply:
changed it to `deploy your application in the cloud as a Docker container`. removed the part about dabl, let's wait until create-react-app actually contains instructions on how to deploy to dabl.

Reply:
also added the reference to the sandbox.